### PR TITLE
Make plonky3 no-std

### PR DESCRIPTION
**Pull request by Schaeff**
uni-stark was no-std, but we removed this in #1857.
This change moves the parts which require std (threads, io) to the `backend` crate and makes the `plonky3` crate no-std.
Also, it simplified the interface of the verifier to only requiring the necessary data (the circuit for each table).

### CODE REVIEW

File: backend/Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,12 @@ halo2_curves = { version = "0.6.1", package = "halo2curves", optional = true}
 snark-verifier = { git = "https://github.com/powdr-labs/snark-verifier", branch = "powdr-halo2", optional = true }
 halo2_solidity_verifier = { git = "https://github.com/powdr-labs/halo2-solidity-verifier", branch = "main", features = ["evm"], optional = true }
 
+p3-commit = { git = "https://github.com/plonky3/Plonky3.git", features = [
+    "test-utils",
+], optional = true }
+p3-matrix = { git = "https://github.com/plonky3/Plonky3.git", optional = true } 
+p3-uni-stark = { git = "https://github.com/plonky3/Plonky3.git", optional = true }
```
Comment (leonardoalt) [inline backend/Cargo.toml:43]: https://github.com/powdr-labs/powdr/pull/1917 changes the branch from main to specific commit, maybe you can already rebase on top of it?

**Review by leonardoalt — COMMENTED** [review]
Nice!

File: backend/Cargo.toml (hunk 1 above)
Comment (Schaeff) [reply backend/Cargo.toml:43]: Ah it's merged! Will fix

**Comment by leonardoalt** [comment]
I thnk this needs a merge from main again

**Review by leonardoalt — APPROVED** [review]
sickkk

**Comment by leonardoalt** [comment]
more conflicts :upside_down_face:

**Comment by leonardoalt** [comment]
conflix :sob:

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**